File: insight/src/main/java/info/nightscout/androidaps/plugins/pump/insight/satl/KeyResponse.java
package info.nightscout.androidaps.plugins.pump.insight.satl;

import info.nightscout.androidaps.plugins.pump.insight.utils.ByteBuf;

public class KeyResponse extends SatlMessage {

    private byte[] randomData;
    private byte[] preMasterSecret;

    @Override
    protected void parse(ByteBuf byteBuf) {
        randomData = byteBuf.readBytes(28);
        byteBuf.shift(4);
        preMasterSecret = byteBuf.getBytes(256);
    }

    public byte[] getRandomData() {
        return this.randomData;
    }

    public byte[] getPreMasterSecret() {
        return this.preMasterSecret;
    }
}

File: insight/src/main/java/info/nightscout/androidaps/plugins/pump/insight/satl/KeyResponse.kt
package info.nightscout.androidaps.plugins.pump.insight.satl

import info.nightscout.androidaps.plugins.pump.insight.satl.SatlMessage
import info.nightscout.androidaps.plugins.pump.insight.utils.ByteBuf

class KeyResponse : SatlMessage() {

    lateinit var randomData: ByteArray
        private set
    lateinit var preMasterSecret: ByteArray
        private set

    override fun parse(byteBuf: ByteBuf) {
        randomData = byteBuf.readBytes(28)
        byteBuf.shift(4)
        preMasterSecret = byteBuf.getBytes(256)
    }
}